From 4f53acbf0e3ba3cb9eb44518e25ba3e923312351 Mon Sep 17 00:00:00 2001
From: Li Wang <li.wang@windriver.com>
Date: Wed, 8 Apr 2015 11:01:24 +0800
Subject: [PATCH 9/9] arch/powerpc/kernel: Address IBM PPC476 erratum 48
 (update)

git://git.yoctoproject.org/linux-yocto-3.4
commit 2991cb14f151762ba0856facf600c18eaf3ee382

This patch updates the fix to PPC476 erratum 48 in order to use the
correct calculation for L2 cache register for CPUs 4 and 5.

Signed-off-by: Sangeetha Rao <sangeetha.rao@intel.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
Signed-off-by: Li Wang <li.wang@windriver.com>
---
 arch/powerpc/kernel/traps.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/powerpc/kernel/traps.c b/arch/powerpc/kernel/traps.c
index fdbe2b2..cfd8f49 100644
--- a/arch/powerpc/kernel/traps.c
+++ b/arch/powerpc/kernel/traps.c
@@ -881,8 +881,8 @@ void machine_check_exception(struct pt_regs *regs)
 		mtdcr((cpu + 3) * 256, 0x304);
 		l2plbstats1 = mfdcr((cpu + 3) * 256 + 4);
 	} else {
-		mtdcr((cpu + 9)*256, 0x304);
-		l2plbstats1 = mfdcr((cpu + 9) * 256 + 4);
+		mtdcr((cpu + 15) * 256, 0x304);
+		l2plbstats1 = mfdcr((cpu + 15) * 256 + 4);
 	}
 
 	if (((mcsr & 0x80200000) == 0x80200000) &&
@@ -894,7 +894,7 @@ void machine_check_exception(struct pt_regs *regs)
 			mtdcr((cpu + 3) * 256 + 4,
 				0x000C0000);
 		} else {
-			mtdcr((cpu + 9) * 256 + 4,
+			mtdcr((cpu + 15) * 256 + 4,
 				0x000C0000);
 		}
 		printk(KERN_INFO "machine_check_exception: Core %d: MCSR=0x%x l2plbstats1=0x%x",
-- 
1.7.5.4

